bundles/org.openhab.automation.pythonscripting/src/main/resources/lib/openhab/helper.py

```diff
@@ -84,16 +86,30 @@ def __init__(self, name=None, description=None, tags=None, triggers=None, condit
         self.conditions = conditions
         self.profile = profile
 
-    def __call__(self, clazz):
+    def __call__(self, clazzOrFunc):
         proxy = self
 
-        class_package = proxy.getClassPackage(clazz.__name__)
+        if isfunction(clazzOrFunc):
+            class_package = clazzOrFunc.__name__
+            _rule_obj = clazzOrFunc
+        else:
+            class_package = proxy.getClassPackage(clazzOrFunc.__name__)
+            _rule_obj = clazzOrFunc()
 
-        clazz.logger = Java_LogFactory.getLogger( "{}.{}".format(LOG_PREFIX, class_package) )
+        clazzOrFunc.logger = Java_LogFactory.getLogger( LOG_PREFIX + "." + file_package + "." + class_package )
 
-        _rule_obj = clazz()
+        if hasattr(_rule_obj, "_triggers"):
+            logger.info("_rule_obj has _triggers")
+            _triggers = _rule_obj._triggers
+        else:
+            _triggers = []
+
+        if hasattr(_rule_obj, "_conditions"):
+            logger.info("_rule_obj has _conditions")
+            _conditions = _rule_obj._conditions
+        else:
+            _conditions = []
```

Comment:
I would not recommend to introduce a third way of adding triggers and conditions to rules.

1. triggers and conditions attributes in rule decorator
2. buildTriggers and buildConditions in rule class